js/parts/Series.js

```diff
@@ -2182,6 +2227,13 @@ null,
         // Insert the series and re-order all series above the insertion
         // point.
         chart.orderSeries(this.insert(chartSeries));
+        // Set options for series with sorting and set data later.
```

Comment:
Can you explain to me why `setData` has to be delayed when we do data sorting?

Reply:
It has to be delayed because of sorting linked series. We need to call first `chart.linkSeries();` and then `chart.setSeriesData();`.

Reply:
What is the use case for sorting linked series? Is it worth the added level of complexity? Can you show us some use cases?

Reply:
Without this functionality sorting for example two series places the points of the first series in a wrong category: http://jsfiddle.net/BlackLabel/3fLjtvaz/

Now we can sort series depending on one main: http://jsfiddle.net/BlackLabel/v4sg82my/

Use cases:

- Dependent sorting of heatmap from samples: http://jsfiddle.net/BlackLabel/74u91kbc/40/
- "Election" demo: http://jsfiddle.net/BlackLabel/0fwo8ezm/

Reply:
Ok, thanks for explaining